indigo_app/forms.py

```diff
@@ -21,29 +20,31 @@ class Meta:
 
     publication_document_file = forms.FileField(required=False)
 
-    # not as simple as this
-    # if there's currently nothing there, upload
-    # if there is, replace
     # also check for hidden field to delete
     def save(self, commit=True):
-        work = super(WorkForm, self).save()
         self.save_publication_document()
+        work = super(WorkForm, self).save(commit)
         return work
 
     def save_publication_document(self):
         pub_doc_file = self.cleaned_data['publication_document_file']
         if pub_doc_file:
-
-            pub_doc = PublicationDocument(work=self.instance, file=pub_doc_file,
-                                          size=pub_doc_file.size, filename=pub_doc_file.name,
-                                          mime_type="application/pdf")
-            try:
+            # if one already exists for this work, update it
+            if hasattr(self.instance, 'publication_document'):
+                PublicationDocument.objects.filter(work=self.instance).update(
```

Comment:
I don't think this will do the right thing on the actual `file` object, because this goes directly to the database. Instead, you'll need to set the attributes individually.

```
self.instance.publication_document.file = pub_doc_file
self.instance.publication_document.size = pub_doc_file.size,

...
```